### PR TITLE
kfunc definition changes

### PR DESCRIPTION
**Pull request by komaljai**
- Change kfunc definition for bpf_p4tc_tbl_read, xdp_p4tc_tbl_read, p4tc_table_entry_create_bpf_params__local, bpf_p4tc_entry_create, xdp_p4tc_entry_create, bpf_p4tc_entry_create_on_miss, xdp_p4tc_entry_create_on_miss,
bpf_p4tc_entry_update, xdp_p4tc_entry_update, bpf_p4tc_entry_delete, bpf_p4tc_entry_delete - https://github.com/p4tc-dev/p4tc-e2e/issues/62
- Added implementation for beXX annotation - https://github.com/p4tc-dev/p4tc-e2e/issues/67

### CODE REVIEW

**Review by maheswari-s — APPROVED** [review]
Looks fine

File: backends/bmv2/common/backend.h [hunk 1]
```diff
@@ -81,15 +81,13 @@ class Backend {
...
 
-/**
-This class implements a policy suitable for the SynthesizeActions pass.
-The policy is: do not synthesize actions for the controls whose names
-are in the specified set.
-For example, we expect that the code in the deparser will not use any
-tables or actions.
-*/
+/// This class implements a policy suitable for the SynthesizeActions pass.
+// The policy is: do not synthesize actions for the controls whose names
```
Comment (psivanup) [inline backends/bmv2/common/backend.h]: a slash is missing here

File: backends/tc/ebpfCodeGen.cpp [hunk 2]
```diff
@@ -596,6 +596,15 @@ void PnaStateTranslationVisitor::compileExtractField(const IR::Expression *expr,
             if (annoVal->text == "macaddr" || annoVal->text == "ipv4" || annoVal->text == "ipv6") {
                 noEndiannessConversion = true;
                 break;
+            } else if (annoVal->text == "be16" || annoVal->text == "be32" ||
+                       annoVal->text == "be64") {
+                std::string sInt = annoVal->text.substr(2).c_str();
+                unsigned int width = std::stoi(sInt);
+                if (widthToExtract != width) {
+                    ::error("Width of the field doesnt match the annotation width. '%1%'", field);
```
Comment (psivanup) [inline backends/tc/ebpfCodeGen.cpp:604]: Use ErrorType, almost all other error calls are using error type.

**Review by psivanup — APPROVED** [review]
Looks fine to me

**Review by Sosutha — COMMENTED** [review]
Errortype can be added for newly added error statements

**Pull request closed without merging**